src/librustc_driver/lib.rs

```diff
@@ -159,12 +159,11 @@ pub use rustc_trans_utils::trans_crate::MetadataOnlyTransCrate as DefaultTransCr
 pub use rustc_trans::LlvmTransCrate as DefaultTransCrate;
 
 #[cfg(not(feature="llvm"))]
-mod rustc_trans {
+pub mod rustc_trans {
```

Comment:
All of this should be moved to the `TransCrate` trait object.

Reply:
Most of it is llvm specific.

Reply:
Still, it should be provided through an opaque interface for maximal flexibility.